tests/containerTests.js

```diff
@@ -34,6 +34,15 @@ describe('Container', () =>  {
             expect(foo.bar).toBe(5);
         });
 
+        it('should register/resolve an object with a custom factory', () =>  {
+            let Foo = function(){};
+            Foo.prototype = createObject({ bar: { value : 5 }});
+            container.registerFactory('foo', () => new Foo);
+            let foo = container.resolve('foo');
+            expect(foo.bar).toBeDefined();
+            expect(foo.bar).toBe(5);
```

Comment:
When creating properties with `Object.create()` you define the getters by defining the `value` of the prop, but then the resulting object is used as any other object. 
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/create#Using_propertiesObject_argument_with_Object.create()